docs/navigation-prop.md

```diff
@@ -6,25 +6,23 @@ sidebar_label: Navigation prop
 
 Each `screen` component in your app is provided with the `navigation` prop automatically. It looks like this:
 
-* `this.props.navigation`
-  * `navigate` - go to another screen, figures out the action it needs to take to do it
+* Navigating
+  * `navigate` - navigates to another screen
   * `goBack` - close active screen and move back in the stack
+  * `replace` - replace the current route with a new one
+  * `dispatch` - manually send a navigation action to the navigator. This is a lower level method that you generally will not need to use as methods like `navigate` and `goBack` will handle it for you.
+* Navigating with stack-based navigators like `StackNavigator`
+  * `push` - pushes a new screen onto a stack navigator and navigates to it
+  * `pop` - pop the current screen off of the stack navigator and navigate to the previous screen
+  * `popToTop` - pop all screens except the bottom-most off of the stack navigator and navigate to that screen
+* State and lifecycle
   * `addListener` - subscribe to updates to navigation lifecycle
   * `isFocused` - function that returns `true` if the screen is focused and `false` otherwise.
   * `state` - current state/routes
   * `setParams` - make changes to route's params
   * `getParam` - get a specific param with fallback
-  * `dispatch` - send an action to router
-
-It's important to highlight the `navigation` prop is _not_ passed in to _all_ components; only `screen` components receive this prop automatically! React Navigation doesn't do anything magic here. For example, if you were to define a `MyBackButton` component and render it as a child of a screen component, you would not be able to access the `navigation` prop on it.
 
-There are several additional functions on `this.props.navigation` that only if the current navigator is a stack navigator. These functions are alternatives to `navigate` and `goBack` and you can use whichever you prefer. The functions are:
```

Comment:
I see what you mean. How about if I put that sentence in the bulleted list above, as I've done in the latest PR? My problem with the current phrasing is I thought I saw that the stack-based functions are added for *all* navigator types, stack or otherwise.